src/classes/LeadRepository.cls

```diff
@@ -30,6 +30,14 @@ public without sharing class LeadRepository extends SObjectRepository {
         return (List<Lead>)Database.query(query);
     }
 
+    public List<Lead> searchInAllFields(String searchTerm) {
+        String query = this
+            .addConditionIsConverted(false)
+            .getSearchQuery(searchTerm);
+
+        return (List<Lead>)Search.query(query)[0];
```

Comment:
Dynamic SOSL should not be used as it could expose the application to SOSL injection security vulnerabilities.